src/backend/distributed/master/worker_node_manager.c

```diff
@@ -414,6 +430,30 @@ ReferenceTablePlacementNodeList(LOCKMODE lockMode)
 }
 
 
+/*
+ * CoordinatorNode returns the WorkerNode object for coordinator node if it is
+ * added to pg_dist_node, otherwise errors out.
+ */
+WorkerNode *
+CoordinatorNode()
+{
+	if (!CoordinatorAddedAsWorkerNode())
+	{
+		ereport(ERROR, (errmsg("couldn't find coordinator node as it is not "
+							   "added to pg_dist_node.")));
+	}
+
+	EnsureModificationsCanRun();
```

Comment:
why do we have this here?

Reply:
I also wonder if we ever need this here. But the reason I put it here is that we call it in other utility functions returning `WorkerNode` objects in `worker_node_manager.c`, I should re-think this :)

Reply:
I don't think we need `EnsureModificationsCanRun();` here